pkg/query/querier_test.go

```diff
@@ -506,105 +575,127 @@ func TestQuerier_Select(t *testing.T) {
-						testutil.Ok(t, r.Err)
-
-						testSelectResponse(t, sc.expected, catcher.resp[0])
-
-						warns := catcher.warns()
-						// We don't care about anything else, all should be recorded.
-						testutil.Assert(t, len(warns) == 1, "expected only single warnings")
-						testutil.Assert(t, len(catcher.resp) == 1, "expected only single response, subqueries?")
-
-						w := warns[0]
+						t.Cleanup(localStoreQ.Close)
+						localStoreR := localStoreQ.Exec(context.Background())
+						testutil.Ok(t, localStoreR.Err)
+						actual = promqlResToSeries(localStoreR)
 						if tcase.expectedWarning != "" {
-							testutil.Equals(t, 1, len(w))
-							testutil.Equals(t, tcase.expectedWarning, w[0].Error())
+							warns := localStoreR.Warnings
+							testutil.Assert(t, len(warns) == 1, "expected only single warnings")
+							testutil.Equals(t, tcase.expectedWarning, warns[0].Error())
 						}
-					})
+					}
+
+					testutil.Equals(t, sc.expected, actual, "promql result doesn't match the expected output")
+					if sc.dedup {
+						testutil.Assert(t, len(actual) == 1, "expected only single response, subqueries?")
+					}
 				})
 			}
 		})
 	}
 }
 
-func testSelectResponse(t *testing.T, expected []series, res storage.SeriesSet) {
-	var series []storage.Series
-	// Use it as PromQL would do, first gather all series.
-	for res.Next() {
-		series = append(series, res.At())
-	}
-	testutil.Ok(t, res.Err())
-	testutil.Equals(t, len(expected), len(series))
-
-	for i, s := range series {
-		testutil.Equals(t, expected[i].lset, s.Labels())
-		samples := expandSeries(t, s.Iterator())
-		expectedCpy := make([]sample, 0, len(expected[i].samples))
-		for _, s := range expected[i].samples {
-			v := s.v
-			if value.IsStaleNaN(v) {
-				// Nan != Nan, so substitute for another value.
-				// This is required for testutil.Equals to work deterministically.
-				v = hackyStaleMarker
-			}
-			expectedCpy = append(expectedCpy, sample{t: s.t, v: v})
+func jsonToSeries(t *testing.T, filename string) []series {
+	file, err := ioutil.ReadFile(filename)
+	testutil.Ok(t, err)
+
+	data := Response{}
+	err = json.Unmarshal(file, &data)
+	testutil.Ok(t, err)
+
+	var sers []series
+	for _, ser := range data.Data.Results {
+		var lbls labels.Labels
+		for n, v := range ser.Metric {
+			lbls = append(lbls, labels.Label{
+				Name:  string(n),
+				Value: string(v),
+			})
 		}
-		testutil.Equals(t, expectedCpy, samples, "samples for series %v does not match", i)
+		// Label names need to be sorted
+		sort.Sort(lbls)
+
+		var smpls []sample
+		for _, smp := range ser.Values {
+			smpls = append(smpls, sample{
+				t: int64(smp.Timestamp),
+				v: float64(smp.Value),
+			})
+		}
+
+		sers = append(sers, series{
+			lset:    lbls,
+			samples: smpls,
+		})
 	}
-}
 
-type mockedQueryable struct {
-	q storage.Querier
+	// Sort the series by their labels.
+	sort.Slice(sers, func(i, j int) bool {
+		return labels.Compare(sers[i].lset, sers[j].lset) <= 0
+	})
+
+	return sers
```

Comment:
Never used before shortcut, let be consistent, maybe `ss` or `s`? or just `series`?